extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php

```diff
@@ -511,17 +511,37 @@ public function blockedPage() {
 			return '';
 		}
 
-		list ($blockerName, $reason, $ip, $blockid, $blockTimestamp, $blockExpiry, $intended) = array(
+		list ($blockerName, $reason, $ip, $blockid, $blockTimestamp, $blockExpiry, $intended, $isGlobal) = array(
 			User::whoIs( $wgUser->blockedBy() ),
 			$wgUser->blockedFor() ? $wgUser->blockedFor() : wfMsg( 'blockednoreason' ),
 			$wgRequest->getIP(),
 			$wgUser->getBlockId(),
 			$wgLang->timeanddate( wfTimestamp( TS_MW, $wgUser->mBlock->mTimestamp ), true ),
 			$wgUser->mBlock->mExpiry,
-			$wgUser->mBlock->mAddress
+			$wgUser->mBlock->mAddress,
+			$wgUser->mBlockedGlobally
 		);
 
-		$blockerLink = '[[' . $wgContLang->getNsText( NS_USER ) . ":{$blockerName}|{$blockerName}]]";
+		// Hide username of blocker if this is a global block (see lines 2112-2129 of includes/Title.php)
+		if ($isGlobal) {
+			$blocker = User::newFromName($blockerName);
+			if ($blocker instanceof User) {
+				// user groups to be displayed instead of user name
+				$groups = [
+					'staff',
+					'vstf',
+				];
+				$blockerGroups = $blocker->getEffectiveGroups();
+
+				foreach($groups as $group) {
+					if (in_array($group, $blockerGroups)) {
+						$blockerLink = wfMessage("group-$group")->plain();
+					}
+				}
+			}
+		} else {
+			$blockerLink = '[[' . $wgContLang->getNsText( NS_USER ) . ":{$blockerName}|{$blockerName}]]";
+		}
```

Comment:
@macbre: this change is already implemented for all other times when block
information is shown to the user - the current implementation does not work
with article / blog comments, though, and this should rectify that
On Jul 27, 2015 3:56 AM, "Maciej Brencz" <notifications@github.com> wrote:

> In extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php
> <https://github.com/Wikia/app/pull/7762#discussion_r35512359>:
>
> > +				// user groups to be displayed instead of user name
> > +				$groups = [
> > +					'staff',
> > +					'vstf',
> > +				];
> > +				$blockerGroups = $blocker->getEffectiveGroups();
> > +
> > +				foreach($groups as $group) {
> > +					if (in_array($group, $blockerGroups)) {
> > +						$blockerLink = wfMessage("group-$group")->plain();
> > +					}
> > +				}
> > +			}
> > +		} else {
> > +			$blockerLink = '[[' . $wgContLang->getNsText( NS_USER ) . ":{$blockerName}|{$blockerName}]]";
> > +		}
>
> Thanks for the fix. LGTM.
>
> While VOLDEV-117 says that we should make this change for comments only, I
> think that this should be consistent across all places where block
> information is presented to the user. Given that let's introduce a global
> helper function (bind to a block-related MW hook?) that will apply this
> logic for all blocks.
>
> @Wikia/community-engineering
> <https://github.com/orgs/Wikia/teams/community-engineering>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Wikia/app/pull/7762/files#r35512359>.
>